lib/ood_core/job/adapters/sge/helper.rb

```diff
@@ -47,6 +52,41 @@ def batch_submit_args(script, after: [], afterok: [], afternotok: [], afterany:
     args
   end
 
+  # @brief      Detect whether script content contains either -cwd or -wd
+  #
+  # @param      content  The script content
+  # 
+  # Examples:
+  #     #$-wd /home/ood/ondemand  # should match
+  #     #$  -wd /home/ood/ondemand  # should match
+  #     #$  -cwd /home/ood/ondemand  # should match
+  #     #$ -j yes -wd /home/ood/ondemand  # should match
+  #     #$ -j yes -o this-wd /home/ood/ondemand  # should NOT match
+  #       #$ -t 1-10:5 -wd /home/ood/ondemand  # should NOT match
+  #
+  # @return     [bool]
+  #
+  def script_contains_wd_directive?(content)
+    content.slice(
+      # Only search within the script's first 1024 characters in case the user is
+      # putting non-line delimited data into their scripts.
+      0, 1024
+    ).split(
+      "\n"
+    ).slice(
+      # Only scan in the first 20 rows
+      0, 20
```

Comment:
Is there any reason to do this now that we do the first 1024 chars?

Reply:
I suppose not.